pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -1758,7 +1758,7 @@ public PreparedStatement prepareStatement(String sql, int @Nullable [] columnInd
     }
 
     checkClosed();
-    throw new PSQLException(GT.tr("Returning autogenerated keys is not supported."),
+    throw new PSQLException(GT.tr("Returning autogenerated keys by column index is not supported."),
```

Comment:
I wonder how we should handle that with regards to `gettext` translations.
It seems none of the current translations include "by column index" though.

Reply:
I see various translations containing that string.

Reply:
So how do we deal with this? In a perfect world we would have an army of translators to update the translations. @petere just to clarify: Are you saying that a number of the translations already use the words "by column index" ?

Reply:
It worries me that we need to stop fixing messages or typos because we need to maintain the translations, that by the way there are quite a few, and probably many of them have not been updated since 15 years ago.

So or we have an army of translators, or we drop some of them and maintain just the ones that can be maintained.

Reply:
> So how do we deal with this? In a perfect world we would have an army of translators to update the translations. @petere just to clarify: Are you saying that a number of the translations already use the words "by column index" ?

I'm a bit confused by this discussion and what the requirements are. The new error message wording I'm proposing is already used in three other places. This change just makes the wordings consistent in all the cases, and the old wording is factually wrong. It appears AFAICT that most of the translations haven't been updated for the newer wording yet, but this is an existing problem. Adding yet another use of the same message doesn't substantially change the situation.